### ISSUE TITLE
Firebase Crash when running unit test "[FBLPromise HTTPBody]: unrecognized selector"

### ISSUE THREAD

**Issue by hivenet-erwanjestin** [firebase/firebase-ios-sdk#11777]
### Description

I have a crash when running a unit test with Crashlytics. It doesn't happen when using the app.

Here is my code:
```
FirebaseApp.configure(options: fileopts)
```

I'm using Firebase Crashlytics only - in a dedicated swift module

Here is the screenshot of the error:

<img width="1293" alt="Capture d’écran 2023-09-06 à 11 44 02" src="https://github.com/firebase/firebase-ios-sdk/assets/124664565/d44cb344-ba88-45c4-b43d-128d75f405f5">

### Reproducing the issue

- Create a new Swift package
- Add a Firebase dependency `.package(url: "https://github.com/firebase/firebase-ios-sdk.git", .upToNextMajor(from: "10.0.0"))`
- Configure Firebase in this package `FirebaseApp.configure(options: fileopts)`
- Call it from an iOS app
- Create a unit test on the iOS app and one empty performance test (with `self.measure {...`)

### Firebase SDK Version

10.9

### Xcode Version

14.3

### Installation Method

Swift Package Manager

### Firebase Product(s)

Crashlytics

### Targeted Platforms

iOS

### Relevant Log Output

```shell
Thread 15: "-[FBLPromise HTTPBody]: unrecognized selector sent to instance 0x60000153f3f0"
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json
{
  "object": {
    "pins": [
      {
        "package": "abseil",
        "repositoryURL": "https://github.com/google/abseil-cpp-binary.git",
        "state": {
          "branch": null,
          "revision": "bfc0b6f81adc06ce5121eb23f628473638d67c5c",
          "version": "1.2022062300.0"
        }
      },
      {
        "package": "AnyCodable",
        "repositoryURL": "https://github.com/Flight-School/AnyCodable",
        "state": {
          "branch": null,
          "revision": "862808b2070cd908cb04f9aafe7de83d35f81b05",
          "version": "0.6.7"
        }
      },
      {
        "package": "Firebase",
        "repositoryURL": "https://github.com/firebase/firebase-ios-sdk",
        "state": {
          "branch": null,
          "revision": "2bfe6abe1014aafe5cf28401708f7d39f9926a76",
          "version": "10.14.0"
        }
      },
      {
        "package": "GoogleAppMeasurement",
        "repositoryURL": "https://github.com/google/GoogleAppMeasurement.git",
        "state": {
          "branch": null,
          "revision": "03b9beee1a61f62d32c521e172e192a1663a5e8b",
          "version": "10.13.0"
        }
      },
      {
        "package": "GoogleDataTransport",
        "repositoryURL": "https://github.com/google/GoogleDataTransport.git",
        "state": {
          "branch": null,
          "revision": "aae45a320fd0d11811820335b1eabc8753902a40",
          "version": "9.2.5"
        }
      },
      {
        "package": "GoogleUtilities",
        "repositoryURL": "https://github.com/google/GoogleUtilities.git",
        "state": {
          "branch": null,
          "revision": "c38ce365d77b04a9a300c31061c5227589e5597b",
          "version": "7.11.5"
        }
      },
      {
        "package": "gRPC",
        "repositoryURL": "https://github.com/google/grpc-binary.git",
        "state": {
          "branch": null,
          "revision": "f1b366129d1125be7db83247e003fc333104b569",
          "version": "1.50.2"
        }
      },
      {
        "package": "GTMSessionFetcher",
        "repositoryURL": "https://github.com/google/gtm-session-fetcher.git",
        "state": {
          "branch": null,
          "revision": "d415594121c9e8a4f9d79cecee0965cf35e74dbd",
          "version": "3.1.1"
        }
      },
      {
        "package": "InteropForGoogle",
        "repositoryURL": "https://github.com/google/interop-ios-for-google-sdks.git",
        "state": {
          "branch": null,
          "revision": "2d12673670417654f08f5f90fdd62926dc3a2648",
          "version": "100.0.0"
        }
      },
      {
        "package": "leveldb",
        "repositoryURL": "https://github.com/firebase/leveldb.git",
        "state": {
          "branch": null,
          "revision": "0706abcc6b0bd9cedfbb015ba840e4a780b5159b",
          "version": "1.22.2"
        }
      },
      {
        "package": "nanopb",
        "repositoryURL": "https://github.com/firebase/nanopb.git",
        "state": {
          "branch": null,
          "revision": "819d0a2173aff699fb8c364b6fb906f7cdb1a692",
          "version": "2.30909.0"
        }
      },
      {
        "package": "Promises",
        "repositoryURL": "https://github.com/google/promises.git",
        "state": {
          "branch": null,
          "revision": "e70e889c0196c76d22759eb50d6a0270ca9f1d9e",
          "version": "2.3.1"
        }
      },
    ]
  },
  "version": 1
}

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by paulb777** [comment]
@hivenet-erwanjestin Thanks for the report. Would you share the repro case in a GitHub repo or zip?

**Comment by kcw-gruntsoftware** [comment]
While this is another bug caused by using Firebase, I was able to resolve the issue and it was just a condition in XCTest that creates a crash.

Followed the guidance in Firebase [documentation](https://firebase.google.com/docs/ios/setup?authuser=0&hl=en#swift_1) by adding -ObjC to the Other Linker Flags (found in Build Settings)

Removed the empty testPerformance method for new XCTest Classes.

The second step was revealed when I added any new XCTest class and the crash was always present. Removing it allowed the tests to run
